blivet/tasks/fswritelabel.py

```diff
@@ -66,7 +66,13 @@ class DosFSWriteLabel(FSWriteLabel):
 
     @property
     def args(self):
-        return [self.fs.device, self.fs.label]
+        if availability.MKDOSFS_NEW_APP.available:
+            if self.fs.label:
+                return [self.fs.device, self.fs.label]
+            else:
+                return [self.fs.device, "--reset"]
+        else:
+            return [self.fs.device, self.fs.label]
```

Comment:
This code is a bit redundant, but it's also clear. It might be better IMO to only have one `return` and maybe just set the second argument in the conditional, defining the first argument either at the top or in the `return`. Your call.